### DISCUSSION TITLE
is this framework usable outside of "roach:run" ?

### DISCUSSION THREAD

**Discussion by michabbb** [roach-php/core#43] [Q&A] [answered]
hi,
i would like to run my own command where i do some stuff after parsing a website.
is there a way to run a spider and fetch results in my own code so for example:

```php
$spider->run();
$items = $spider->getProcessedItems();
```
or is the only way to run "roach:run myspider" as first command that stores the items somewhere and after that I run my own stuff that does something with that items?

i am aware, that there are "ItemProcessors" but when testing that package here I thought, it would be possible to implement the "crawling" somewhere in the middle of my own code.

thanks for any help!

**Comment by ksassnowski** [comment, marked as answer]
I think `Roach::collectSpider` is what you're looking for: https://roach-php.dev/docs/spiders#retrieving-scraped-items-after-a-run

**Reply by michabbb** [reply]
thanks! that seems to be exactly what I have missed, nice. will try 😊
